modules/javafx.graphics/src/main/native-glass/win/ThemeSupport.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "common.h"
+#include "ThemeSupport.h"
+#include "RoActivationSupport.h"
+#include <windows.ui.viewmanagement.h>
+
+using namespace Microsoft::WRL;
+using namespace ABI::Windows::UI;
+using namespace ABI::Windows::UI::ViewManagement;
+
+ThemeSupport::ThemeSupport(JNIEnv* env) : env_(env), mapClass_((jclass)env->FindClass("java/util/Map"))
+{
+    putMethod_ = env->GetMethodID(mapClass_, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+}
+
+ThemeSupport::~ThemeSupport() {
+    env_->DeleteLocalRef(mapClass_);
+}
+
+void ThemeSupport::queryHighContrastScheme(jobject properties)
```

Comment:
@mstr2 Do you think it is possible to use here `Windows.UI.ViewManagement.AccessibleSettings` instead, same as you do in `queryWindows10ThemeColors` with `Windows.UI.ViewManagement.UISettings`?

I believe `Windows.UI.ViewManagement.AccessibleSettings.HighContrastScheme` contains a not localised string for the high contrast scheme, and that would allow solving https://bugs.openjdk.java.net/browse/JDK-8185447.

Reply:
I just checked on Windows 10 v20H2 (build 19042.985), and unfortunately the `Windows.UI.ViewManagement.AccessibleSettings.HighContrastScheme` contains a localized name for the high contrast scheme.

Reply:
Oh, too bad, but thanks for testing anyway.
Do you know if there is such API that returns a non-localized string? I read it somewhere...

Reply:
I don't know of such an API. Even when I change the locale of the current thread to English, the API returns the localized (non-English) name of the scheme.

Reply:
Yes, I know, I've got a possible fix for JDK-8185447 based precisely in using the localized high contrast scheme names (limited to the same languages supported in Controls).

Reply:
Just for reference: I've proposed [this enhancement](https://bugs.openjdk.java.net/browse/JDK-8267554) to load dynamic stylesheets that could be injected with the actual scheme colors. If added, this might be a better fix than matching color scheme names.

Reply:
How would that help? How would the high contrast stylesheets be loaded?

Reply:
Instead of hardcoding the colors in stylesheets such as Modena's `blackOnWhite.css`, you could use placeholder symbols inside of the stylesheet. Then, you could read the stylesheet into a string, replace the placeholders with the actual system colors, and load the modified stylesheet with a "data" URI.